modules/node-group/templates/zero_storagepool.service.tmpl

```diff
@@ -0,0 +1,21 @@
+[Unit]
+Description=Zero storagepool partition on first boot
+# Conditions copied from upstream machine-config-daemon-firstboot.service
+# Make sure it runs only on OSTree booted system
+ConditionPathExists=/run/ostree-booted
+# Removal of this file signals firstboot completion
+ConditionPathExists=/etc/ignition-machine-config-encapsulated.json
+# Use same conditions as for machine-config-daemon-firstboot.service
+After=machine-config-daemon-pull.service
+Before=crio.service crio-wipe.service
+Before=kubelet.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=dd if=/dev/zero of=${partition} bs=1M count=${size_mb}
```

Comment:
Isn't there a less timeconsuming way to do this? From my personal experiments with Ceph, I have something in mind that took a -zap option. However I can not find it anymore.

If we stick with `dd`, is the `count=` option required? Won't `dd` exit when the end of the partition is reached?

Reply:
I couldn't find anything useful in the Rook.io documentation regarding cleaning partitions when bootstrapping the cluster. Directly controlling the behavior of `ceph-volume` and other Ceph tooling is not really possible with Rook, as the operator takes care of all of the nitty-gritty details for you.

Note that the `dd` took roughly 1 minute for 180GB when I tested it yesterday, and it runs in parallel with the initial machine-config bootstrap, so I'm not too concerned about the delay during initial creation of the VM, especially since the maximum partition size which we need to zero is roughly 700GB.

The count option is added to avoid `dd` exiting with exit code 1, marking the unit as failed, which is what happens without the count option when the end of the partition is reached.

As discussed in the commit message, it probably would be sufficient to zero the first couple GB of the partition to make Rook/Ceph happy, but I've erred on the side of caution with the current implementation.